### PR TITLE
show breakpoints in designview

### PR DESCRIPTION
**Pull request by pahans**

### CODE REVIEW

File: modules/web/js/ballerina/ast/node.js [hunk 1]
```diff
@@ -428,6 +428,14 @@ class ASTNode extends EventChannel {
         }
     }
 
+    addBreakpoint() {
+        this.isBreakpoint = true;
+    }
+
+    removeBreakpoint() {
```
Comment (hemikak) [inline modules/web/js/ballerina/ast/node.js]: Better to have doc comments here.

File: modules/web/js/ballerina/components/breakpoint.jsx [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React from "react";
+import ImageUtil from './image-util';
+import PropTypes from 'prop-types';
+
+class Breakpoint extends React.Component {
```
Comment (hemikak) [inline modules/web/js/ballerina/components/breakpoint.jsx]: Better to have class comments here.

File: modules/web/js/ballerina/components/breakpoint.jsx [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+  constructor() {
+      super();
+      this.state = {
+          canShowRemove: false
+      }
+  }
+
+  showRemoveBreakpointIcon(isMouseOver) {
```
Comment (hemikak) [inline modules/web/js/ballerina/components/breakpoint.jsx]: Better to have doc comments here.

File: modules/web/js/ballerina/components/statement-decorator.jsx [hunk 4]
```diff
@@ -71,6 +73,31 @@ class StatementDecorator extends React.Component {
...
+
+	onBreakpointClick() {
+			const { model } = this.props;
+			const { isBreakpoint = false } = model;
+			if(model.isBreakpoint) {
+					model.removeBreakpoint();
+			} else {
+					model.addBreakpoint();
+			}
+	}
```
Comment (hemikak) [inline modules/web/js/ballerina/components/statement-decorator.jsx]: Theres a formatting issue here.

File: modules/web/js/ballerina/visitors/find-breakpoint-lines-visitor.js [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import ASTVisitor from './ast-visitor';
+import _ from 'lodash';
+
+class FindBreakpointLinesVisitor extends ASTVisitor {
```
Comment (hemikak) [inline modules/web/js/ballerina/visitors/find-breakpoint-lines-visitor.js]: Better to have class comments.

File: modules/web/js/ballerina/components/statement-decorator.jsx (hunk 4 above)
Comment (pahans) [inline modules/web/js/ballerina/components/statement-decorator.jsx]: fixed with https://github.com/ballerinalang/composer/pull/2012

File: modules/web/js/ballerina/visitors/find-breakpoint-lines-visitor.js (hunk 5 above)
Comment (pahans) [inline modules/web/js/ballerina/visitors/find-breakpoint-lines-visitor.js]: fixed with https://github.com/ballerinalang/composer/pull/2011

File: modules/web/js/ballerina/components/breakpoint.jsx (hunk 3 above)
Comment (pahans) [inline modules/web/js/ballerina/components/breakpoint.jsx]: fixed with https://github.com/ballerinalang/composer/pull/2013

File: modules/web/js/ballerina/components/breakpoint.jsx (hunk 2 above)
Comment (pahans) [inline modules/web/js/ballerina/components/breakpoint.jsx]: fixed with https://github.com/ballerinalang/composer/pull/2013

File: modules/web/js/ballerina/ast/node.js (hunk 1 above)
Comment (pahans) [inline modules/web/js/ballerina/ast/node.js]: fixed with https://github.com/ballerinalang/composer/pull/2013

**Pull request merged**